pkg/resources/service/yaml-manifest.go

```diff
@@ -79,7 +79,7 @@ func (s *Service) DeployFunctions(functions []Service, removeOrphans bool, threa
 		}
 	}
 
-	if removeOrphans {
+	if removeOrphans && !client.Dry {
```

Comment:
Another solution would be to disable all interactions with a cluster in `removeOrphans()` function, but there is not a much value in moving this conditional there because without stubbing http requests to a cluster we can't test removing orphaned functions anyway.